pawn/Driver/Driver.pwn

```diff
@@ -224,12 +259,90 @@ public OnPlayerKeyStateChange(playerid, newkeys, oldkeys) {
             if (VehicleModel(modelId)->isNitroInjectionAvailable())
                 AddVehicleComponent(vehicleId, 1010);
         }
+
+        new bool: rightBlinker = g_blinkerObjects[playerid][0] != 0;
+        new bool: leftBlinker = g_blinkerObjects[playerid][2] != 0;
+        // Vehicle keys (q): blinker left
+        if (PRESSED(VEHICLE_KEYS_BINDING_BLINKER_RIGHT) && 
+            (vehicleKeys & VEHICLE_KEYS_BLINKER_RIGHT)) {
+
+            rightBlinker = !rightBlinker;
+        }
+
+        // Vehicle keys (e): blinker right
+        if (PRESSED(VEHICLE_KEYS_BINDING_BLINKER_LEFT) && 
+            (vehicleKeys & VEHICLE_KEYS_BLINKER_LEFT)) {
+
+            leftBlinker = !leftBlinker;
+        }
+
+        SetBlinker(playerid, vehicleId, leftBlinker, rightBlinker);
     }
 
     LegacyPlayerKeyStateChange(playerid, newkeys, oldkeys);
     return 1;
 }
 
+// Enable the |left| and or |right| blinkers for |playerid| in |vehicleId|
+SetBlinker(playerid, vehicleId, bool:left, bool:right) {
+    new const blinkerModel = 19294;
+    new const modelId = GetVehicleModel(vehicleId);
+
+    if(IsModelBoat(modelId) || IsModelPlane(modelId)) {
+        return;
+    }
+
+    new Float:sizeX, Float:sizeY, Float:sizeZ;
+    GetVehicleModelInfo(modelId, VEHICLE_MODEL_INFO_SIZE, sizeX, sizeY, sizeZ);
+
+    if (right) {
+        if (g_blinkerObjects[playerid][0] == 0) {
+            g_blinkerObjects[playerid][0] = CreateObject(blinkerModel, 0, 0, 0, 0, 0, 0, 0);
+            AttachObjectToVehicle(g_blinkerObjects[playerid][0], vehicleId, sizeX/2.23, sizeY/2.23, 
+                0.1, 0, 0, 0);
+
+            g_blinkerObjects[playerid][1] = CreateObject(blinkerModel, 0, 0, 0, 0, 0, 0, 0);
+            AttachObjectToVehicle(g_blinkerObjects[playerid][1], vehicleId, sizeX/2.23, -sizeY/2.23, 
+                0.1, 0, 0, 0);
+        }
+    } else {
+        DestroyBlinkerObject(playerid, 0);
+        DestroyBlinkerObject(playerid, 1);
+    }
+
+    if (left) {
+        if (g_blinkerObjects[playerid][2] == 0) {
+            g_blinkerObjects[playerid][2] = CreateObject(blinkerModel, 0, 0, 0, 0, 0, 0, 0);
+            AttachObjectToVehicle(g_blinkerObjects[playerid][2], vehicleId, -sizeX/2.23, sizeY/2.23, 
+                0.1, 0, 0, 0);
+
+            g_blinkerObjects[playerid][3] = CreateObject(blinkerModel, 0, 0, 0, 0, 0, 0, 0);
+            AttachObjectToVehicle(g_blinkerObjects[playerid][3], vehicleId, -sizeX/2.23, -sizeY/2.23, 
+                0.1, 0, 0, 0);
+        }
+    } else {
+        DestroyBlinkerObject(playerid, 2);
+        DestroyBlinkerObject(playerid, 3);
+    }
+}
+
+// This resets the whole blinking status and removes the objects.
+forward StopBlinking(playerid);
+public StopBlinking(playerid) {
```

Comment:
nit: no need for this to be public as far as I can see

Reply:
It does have to be public right?
To disable on "PlayerSyncedData"

Reply:
Nah, read the `public` keyword (when not in a Pawn class) as `exported`, it makes it available to code outside of Pawn, e.g. filterscripts and our JavaScript plugin.

The `PlayerSyncedData` file is part of our Pawn code, so that can access anything else in non-class Pawn just fine. That's actually very similar to how TypeScript and PHP work.

Reply:
Aight, will update. Thanks :)